src/reference-manual/types.Rmd

```diff
@@ -1386,6 +1386,43 @@ As with other assignments, promotions will happen if necessary (of
 `int` to `real` and of `real` to `complex`, along with the
 corresponding container type promotions).
 
+### Unpacking assignment of tuples {-}
+
+For convience of using values stored in tuples, Stan supports
+"unpacking" (or "destructuring") of tuples in an assignment statement.
+
+Given a tuple `T` of type `tuple(T1, ..., Tn)` and a sequence of
+variables of types `v1`, ..., `vn`, where each `vi` has a type which is
+[assignable](#assignment-typing) from type `Ti`, individual elements of
+the tuple may be assigned to the corresponding variables in the sequence
+by the statement
+
+```stan
+(v1, /*...*/, vn) = T;
+```
+
+For example, if `T` is a tuple of type `tuple(int, real, complex)`,
+then the program
+
+```stan
+int i;
+real r;
+complex z;
+
+(i, r, z) = T;
+```
+
+Assigns the result of `T.1` to `i`, the result of `T.2` to `r`, and
+the result of `T.3` to `z`.
+
+These unpacking assignments can be nested if the tuple on the right hand
+side contains nested tuples.
+
+The left hand side of an unpacking assignment must contain only
```

Comment:
This seems to say it's general, but the earlier description says it's just variables.  Which is it?  And if it's indexed, how do we avoid aliasing or check when it's now a runtime issue?

Reply:
Updated the earlier description. See above for how we avoid aliasing, but the short version is the statement that the same variable may not appear more than once is correct. This is a bit overly-conservative, but does prevent the aliasing problem for indexed assignments appearing here.